wazo_calld/plugins/transfers/state.py

```diff
@@ -623,13 +622,15 @@ def transferred_hangup(self):
     def initiator_hangup(self):
         try:
             self._unhold_transferred_call()
-        except ARINotFound:
-            raise TransferCompletionError(self.transfer.id, 'transferred party hung up')
-
-        try:
             self._ari.channels.ring(channelId=self.transfer.transferred_call)
         except ARINotFound:
-            raise TransferCompletionError(self.transfer.id, 'transferred party hung up')
+            logger.info('transferred party hung up while handling initiator hangup')
+            if self.transfer.recipient_call:
```

Comment:
IMHO this condition is unnecessary: we are already in the state Ringback, so the destination call must exist